src/sliding_sync.rs

```
@@ -940,23 +957,71 @@ async fn async_main_loop(
+                                    error!("Login failed: {e:?}");
+                                    Cx::post_action(LoginAction::LoginFailure(e.to_string()));
+                                    enqueue_rooms_list_update(RoomsListUpdate::Status {
+                                        status: e.to_string(),
+                                    });
+                                }
+                            }
+                        }
+                        LoginRequest::LoginBySSO(id) => {
+                            Cx::post_action(LoginAction::SsoPending(true));
+
+                            let cli_parse_result = Cli::try_parse();
+                            let cli = cli_parse_result.unwrap_or(Cli::default());
+                            let (client, client_session) =
+                                build_client(&cli, app_data_dir()).await?;
+                            match client
+                                .matrix_auth()
+                                .login_sso(|sso_url: String| async move {
+                                    let _ = Uri::new(&sso_url).open();
+                                    Ok(())
+                                })
+                                .identity_provider_id(&id)
+                                .await
+                            {
+                                Ok(res) => {
+                                    log!("Logged in successfully? {:?}", client.logged_in());
+                                    enqueue_rooms_list_update(RoomsListUpdate::Status {
+                                        status: format!(
+                                            "Logged in as {:?}. Loading rooms...",
+                                            &res.user_id
+                                        ),
+                                    });
+                                    let _ = persistent_state::save_session(&client, client_session)
+                                        .await;
```

don't silence errors or get lazy about error handling.

---

fixed